_tools/TicketDescriptions.py

```diff
@@ -75,6 +90,46 @@ def translate_layer_desc(item, file_name):
     
     return 0
 
+def get_type_restrictions(item):
+    types = "a"
+
+    if "：ヒト型" in item["jp_explain"] and "/キャスト" not in item["jp_explain"]:
+        types = "h"
+    elif "：キャスト" in item["jp_explain"]:
+        types = "c"
+        
+    if "タイプ1<c>" in item["jp_explain"]:
+        types += "1"
+    elif "タイプ2<c>" in item["jp_explain"]:
+        types += "2"
+    
+    return types
+
+def translate_nlayer_desc(item, file_name):
+    if item["tr_text"] == "": # No name to put in description
+        return -1
+    
+    elif item["tr_explain"] != "" and REDO_ALL == False: # Description already present, leave it alone
+        return -2
+    
+    # Some items are locked to one race and/or type.
+    types = get_type_restrictions(item)
+
+    # Some items hide your innerwear (these are mostly swimsuits).
+    hideinner = False
+    if "着用時はインナーが非表示になります。" in item["jp_explain"]:
+        hideinner = True
+
+    # Translate the description.
+    item["tr_explain"] = (ndesc_formats[LANG] + "{hidepanties}").format(
+        itype = layered_wear_types[item["tr_text"].split("[", )[1][0:2]][LANG] if item["tr_text"].endswith("]")
+                # Exception for default layered wear since it doesn't have [In], [Ba] etc
+                else layered_wear_types[file_name.split("_")[0][0:2]][LANG],
+        typelock = "" if types == "a" else "\n<yellow>※Type: {0}<c>".format(ntype_locks[types][LANG]),
```

Comment:
You'll need a Russian version of "Type" here too, but I'll need to change something on my end first.